index.js

```diff
@@ -32,6 +32,38 @@ function getMainArgs() {
   return process.argv.slice(2);
 }
 
+function setOptionValue(parseOptions, option, value, result) {
+  const multiple = parseOptions.multiples &&
+    parseOptions.multiples.includes(option);
+  const withValue = parseOptions.withValue &&
+    parseOptions.withValue.includes(option);
+
+  // Normal flag: !withValue && value === undefined
+  // Normal value, withValue && value !== undefined
+  // Special case: withValue && value === undefined
+  //    store as normal for withValue with value undefined
+  // Special case: !withValue && value !== undefined
+  //    store as normal for withValue (and not a flag)
+
+  // Flags
+  // Only mark flags for plain flag without a value, expected or otherwise.
+  const isFlag = !withValue && value === undefined;
+  if (isFlag)
+    result.flags[option] = true;
+
+  // Values
+  if (multiple) {
+    // Always store value in array, including for flags.
+    const val = isFlag ? true : value;
+    if (result.values[option] !== undefined)
```

Comment:
Should this be testing for presence, rather than undefined?

Reply:
I was using the undefined test as a casual presence test. The option can't be present with the value `undefined` due to the fact it is a multiple, so I had considered and eliminated that possibility.

However, I thought about it when writing the code, and you asked too. Assuming you agree with my logic, I should add a comment to clarify. (Can cover both questions in one comment.)

Reply:
Added comment.